eth_interface.md

```diff
@@ -19,11 +19,13 @@ We also define the following WebAssembly data types:
 
 ## useGas
 
-Subtracts an amount to the gas counter
+Reduces the gas left counter by an amount.
```

Comment:
Is the "gas left counter" kept outside of the WASM?

Reply:
Yes.

Reply:
I see here one problem, the `takeGas` function does not provide any reference to the current VM execution context. How would a VM handle this in concurrent environment?

Reply:
There are no references in any of the methods. Each contract has its own instance, it is an implementation detail in VM how to add the context to the methods.

Reply:
How is Hera doing it?

Reply:
There's an instance of the methods for each execution (follows the sample way binaryen gives). It can be optimised though, but shouldn't affect the interface.